tests/Data/Test_Contract.cpp

```diff
@@ -0,0 +1,121 @@
+/**
+* Copyright (c) 2018 Zilliqa 
+* This source code is being disclosed to you solely for the purpose of your participation in 
+* testing Zilliqa. You may view, compile and run the code for that purpose and pursuant to 
+* the protocols and algorithms that are programmed into, and intended by, the code. You may 
+* not do anything else with the code without express permission from Zilliqa Research Pte. Ltd., 
+* including modifying or publishing the code (or any part of it), and developing or forming 
+* another public or private blockchain network. This source code is provided ‘as is’ and no 
+* warranties are given as to title or non-infringement, merchantability or fitness for purpose 
+* and, to the extent permitted by law, all liability for your use of the code is disclaimed. 
+* Some programs in this code are governed by the GNU General Public License v3.0 (available at 
+* https://www.gnu.org/licenses/gpl-3.0.en.html) (‘GPLv3’). The programs that are governed by 
+* GPLv3.0 are those programs that are located in the folders src/depends and tests/depends 
+* and which include a reference to GPLv3 in their program files.
+**/
+
+#include "common/Constants.h"
+#include "depends/common/CommonIO.h"
+#include "libCrypto/Schnorr.h"
+#include "libCrypto/Sha2.h"
+#include "libData/AccountData/Account.h"
+#include "libData/AccountData/AccountStore.h"
+#include "libData/AccountData/Transaction.h"
+#include "libPersistence/ContractStorage.h"
+#include "libUtils/DataConversion.h"
+#include "libUtils/Logger.h"
+#include <array>
+#include <string>
+#include <vector>
+
+#include "CrowdFundingCodes.h"
+
+#define BOOST_TEST_MODULE contracttest
+#define BOOST_TEST_DYN_LINK
+#include <boost/test/unit_test.hpp>
+
+using namespace boost::multiprecision;
+using namespace std;
+
+BOOST_AUTO_TEST_SUITE(contracttest)
+
+Address fromAddr;
+Address toAddress;
+KeyPair sender;
+uint256_t nonce = 0;
+
+// Create Transaction to create contract
+BOOST_AUTO_TEST_CASE(testContract)
+{
+    INIT_STDOUT_LOGGER();
+
+    LOG_MARKER();
+
+    AccountStore::GetInstance().Init();
+
+    sender = Schnorr::GetInstance().GenKeyPair();
+
+    std::vector<unsigned char> vec;
+    sender.second.Serialize(vec, 0);
+    SHA2<HASH_TYPE::HASH_VARIANT_256> sha2;
+    sha2.Update(vec);
+    const std::vector<unsigned char>& output = sha2.Finalize();
+
+    copy(output.end() - ACC_ADDR_SIZE, output.end(),
+         fromAddr.asArray().begin());
```

Comment:
can we use this to derive the address from public key instead of reinventing it? https://github.com/Zilliqa/Zilliqa/blob/d3e41248d0f1ddc6f69d5143d020c12310aab153/src/libData/AccountData/Account.h#L115